### ISSUE TITLE
Add chromecast cast support

### ISSUE THREAD

**Issue by JSteunou** [abarisain/dmix#697]
I know the wishlist is already quite big, but if by any chance you could consider this request, I'll be very happy.

**Comment by thwatson** [comment]
+1  I would help test and gather logs.

**Comment by abarisain** [comment]
Alright, I meant to answer this earlier but I couldn't. 

I already toyed with Chromecast support, but the branch is deleted, with good reason.
Here is where I'm probably not going to add that back : 
 - It's a lot of work, really.  
 - It makes MPDroid depend on the play services, which I don't want after thinking about it for quite some time.  
 - It's hard. I managed to hook up a well-configured MPD stream output, but the Chromecast never reconnected on stream drop, and never updated cover art once downloaded, or song title changes. It was a long time ago, but I think that it really needed a server side component to work optimally. I find the chromecast API pretty bugged and hard to understand.  

It's technically possible though, if you don't account the UX, we had a working POC (really too dirty to be merged into develop)

**Comment by JSteunou** [comment]
hey no problem @abarisain I guessed it is a loooot of work, just throwing the idea up. Someone else might take this feature up one day.

In the meantime, I kind of succeed to do it with a nodejs webapp, hosted on the same server my mdp. I build it around [chromecast-player](https://github.com/xat/chromecast-player). I have to play with 2 apps on the tablet, (yours + mine) but it does the job.

This player also has a CLI named [castnow](https://github.com/xat/castnow), so one can cast manually it's mpd stream to a chromecast, it works very well, just add the `--type` option.

**Comment by abarisain** [comment]
Woah, castnow is a nice project !

**Comment by JSteunou** [comment]
yep, thanks korben who blogged about it (maybe you know it as a french ;) )

**Comment by jkaberg** [comment]
Just to add more fuel to the "fire" ;-)

https://github.com/dz0ny/leapcast

**Comment by abarisain** [comment]
Isn't the project pretty much dead ?
Le jeu. 26 mars 2015 à 20:40, Joel Kåberg <notifications@github.com> a
écrit :
